lib/cn/kvs_mblk_desc.c

```diff
@@ -78,8 +78,13 @@ mblk_madvise(const struct kvs_mblk_desc *md, size_t off, size_t len, int advice)
     if (len == 0)
         return 0;
 
-    assert(off + len <= md->wlen_pages * PAGE_SIZE);
     assert(md->map_base);
+    assert(off < md->wlen_pages * PAGE_SIZE);
+
+    if (off + len > md->wlen_pages * PAGE_SIZE)
+        len = md->wlen_pages * PAGE_SIZE - off;
```

Comment:
After 3.0, I would like to change this to an abort() for IMP replica set test, and see if anybody hits it.

@bhaveshvasandani

Reply:
I would actually make the off check a live check:
if (len == 0 || off >= wlen)
    return 0;

Reply:
If someone is passing an offset greater than the length of the block, we have pretty large issues and shouldn't ignore it

Reply:
Fair point, but you also shouldn't use it in a release build if it's out of bounds.  Depending upon the advice things could go awry in a very bad way...

Reply:
This goes back to discussion after discussion we've had of assert() vs EBUG. cc @alexttx

Reply:
Agree with Greg on making off >= wlen a live check. I would return EINVAL if this check fails and leave it up to the caller to handle it.